master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -394,23 +433,32 @@ private[celeborn] class Master(
       // keep it for compatible reason
       context.reply(ReleaseSlotsResponse(StatusCode.SUCCESS))
 
-    case requestSlots @ RequestSlots(_, _, _, _, _, _, _, _, _, _, _) =>
-      logTrace(s"Received RequestSlots request $requestSlots.")
-      executeWithLeaderChecker(context, handleRequestSlots(context, requestSlots))
+    case requestSlots @ RequestSlots(applicationId, _, _, _, _, _, _, _, _, _, _) =>
+      if (checkAuthStatus(applicationId, context)) {
+        // TODO: [CELEBORN-1261]
+        logTrace(s"Received RequestSlots request $requestSlots.")
+        executeWithLeaderChecker(context, handleRequestSlots(context, requestSlots))
+      }
 
     case pb: PbUnregisterShuffle =>
       val applicationId = pb.getAppId
       val shuffleId = pb.getShuffleId
       val requestId = pb.getRequestId
-      logDebug(s"Received UnregisterShuffle request $requestId, $applicationId, $shuffleId")
-      executeWithLeaderChecker(
-        context,
-        handleUnregisterShuffle(context, applicationId, shuffleId, requestId))
+      if (checkAuthStatus(applicationId, context)) {
+        // TODO: [CELEBORN-1261]
+        logDebug(s"Received UnregisterShuffle request $requestId, $applicationId, $shuffleId")
+        executeWithLeaderChecker(
+          context,
+          handleUnregisterShuffle(context, applicationId, shuffleId, requestId))
+      }
 
     case ApplicationLost(appId, requestId) =>
-      logDebug(
-        s"Received ApplicationLost request $requestId, $appId from ${context.senderAddress}.")
-      executeWithLeaderChecker(context, handleApplicationLost(context, appId, requestId))
+      if (checkAuthStatus(appId, context)) {
```

Comment:
`ApplicationLost` is triggered by both `LifecycleManager` (when unregister shuffle) and `Master` (when timeout dead applications). I guess we don't need to check for the later one? We can check whether senderAddress equals to rpcEnv's address.